### ISSUE TITLE
MD033 - false positive in inline code

### ISSUE THREAD

**Issue by pavelpulec** [markdownlint/markdownlint#455]
## Description

Mdl marks inline HTML inside a multiline inline code. 

### Environment

```shell
$ uname -a
Linux computer 6.2.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 20 Apr 2023 16:11:55 +0000 x86_64 GNU/Linux
```

**MDL Version**

```shell
$ mdl --version
0.12.0
```

### Expected Behavior

Anything can be inside a code block, including HTML tags.

### Actual Behavior

I'm getting `MD033 Inline HTML` even when there is only an inline code over multiple lines. See an example below.

## Replication Case

Here is an example code that fails:

```markdown
# header

`<p><px> whatever whatever whatever whatever whatever whatever whatever whatever
<p>`
```

And the actual problem when running `mdl`:

```shell
$ mdl .
./example.md:3: MD033 Inline HTML
./example.md:4: MD033 Inline HTML
```

Try to change `<px>` to `<p>` and the issue is gone. Or if you put this code on one line, then it doesn't hit that bug. This behavior is very strange.

**Comment by nbehrnd** [comment]
@pavelpulec It is the introduction of the line break which triggers the errors; remove it, and they are gone.

You then have a line longer than permitted by the default threshold of 80 characters/line (checked by rule MD013).  Options available to you include

- you refactor the code.  A change of the spaces per indentation level with a good editor, or a proper syntax reformatter written for html may help.  

  Else, wrap the the lines e.g., `fold -s input.md > output.md` at the last white space prior to 80 characters per line.  Because `fold` with the optional `-s` introduces trailing spaces, a `sed -i 's/ *$//' output.md` helps to tidy up.  But since the two process the file without the specifics of .html in mind, the source code can become dysfunctional.

- you define a style file like
  
  ```style.txt
  # Enable all rules by default
  all
 
  rule 'MD013', :line_length => 120
  ```

  and subsequently run `mdl input.md -s ./style.txt` to explicitly change the default threshold in mdl.  (`fold` includes the optional `-w` flag to adjust the width of the columns to other than 80 characters.)

(The call `mdl input.md -r ~MD013` to mute the rule for any lines in the document is possible, too.)
